### ISSUE TITLE
Initial plugin suggestions

### ISSUE THREAD

**Issue by cbruno10** [theapsgroup/steampipe-plugin-keycloak#16]
Thanks @graza-io for another great plugin! I took a look today and left some suggestions/questions below, please let me know if you have any questions on them.

Config file:
- Can you please change `baseurl` to `base_url` to follow the standard snake casing?

Docs:
- Is the brand color `#FCA121` correct? It seems to be more orange, not sure if that was copy and paste from another plugin?
- In `keycloak_user.md`, can you please break up the middle examples into individual examples, e.g., `Get a specific user by ID`, `Get a specific user by username`, `Get a specific user by email`? This allows users to copy and paste as the hub site has a copy icon per code block (in the top right of each code block).
- Several table docs, like for `keycloak_client` and `keycloak_group`, are light on examples. Can you please add a few more so each table doc has at least 3-4 examples? These really help users get started and jump right into using your plugin.

General:
- Is paging supported by the API, and if so does the SDK already handle this?I took a quick look around in the SDK and didn't spot anything.
- Does the API support passing in a limit/max number of results? For APIs that we do, we usually recommend setting that to the passed in `limit` clause if given and if it's lower than the default paging size or limit, e.g., https://github.com/turbot/steampipe-plugin-tfe/blob/main/tfe/table_tfe_team.go#L48-L59, as this will help reduce the number of results returned (helpful especially for large APIs).

Tables:
- I see a few TODOs around adding more optional key quals, like in the `keycloak_user` table. Are you planning on adding these before publishing the initial version?
- Some tables seem like they don't have a lot of columns, like the `keycloak_group` and `keycloak_user`. Are there more columns we can add to both? For instance, for the `keycloak_user` table, in the [SDK](https://github.com/Nerzal/gocloak/blob/main/models.go#L169) and [API](https://www.keycloak.org/docs-api/15.0/rest-api/index.html#_userrepresentation), it seems like there are more potential columns we can add based on what's returned by the API, but I don't see those in the table definition.
- Possibly related to the item above, I did notice that in some tables you pass `BriefRepresentation` as `true`. Is there a reason these tables are only fetching this form of information? I only quickly read over the API docs, so I'm not very familiar with it yet, but for these tables, is it better to have the brief or full representations?

**Comment by graza-io** [comment]
@cbruno10 thanks for this, it really helps get things in shape!

Config File:
- I've addressed swapping `baseurl` for `base_url`.

Docs:
- Have corrected the brand colour to #33C6E9 
- Examples in `keycloak_user.md` have been separated so that they'll be individual code blocks when published.
- I have expanded table docs to include more examples.

General:
- Paging is only supported in API/SDK on the get users endpoint - which I had implemented.
- Have enhanced the above paging to use the limit to reduce pageSize and bow out when context is cancelled (manually/limit hit) or continue for all pages when not.

Tables:
- The `TODO` comments were addressed in #13 I merely forgot to remove the comments.
- This is by design for the initial implementation, when attempting to obtain fuller datasets on these tables I was hit by constant errors - I have these aside to investigate if these are caused by our instance(s) of Keycloak, the SDK or the API. However the most commonly needed data is returned and once we can iron out the kinks in the other data I intend to include this in a future release.
